Adding context to build_after (#7713)
fishtown-analytics/docs.getdbt.com: website/docs/reference/model-configs.md

    [+](/reference/resource-configs/plus-prefix)[sql_header](/reference/resource-configs/sql_header): <string>
    [+](/reference/resource-configs/plus-prefix)[on_configuration_change](/reference/resource-configs/on_configuration_change): apply | continue | fail #only for materialized views on supported adapters
    [+](/reference/resource-configs/plus-prefix)[unique_key](/reference/resource-configs/unique_key): <column_name_or_expression>

>     [+](/reference/resource-configs/plus-prefix)[build_after](/reference/resource-configs/build-after): <dict> # Only for state aware orchestration in the dbt plaftorm 

Author: ```suggestion
    [+](/reference/resource-configs/plus-prefix)[build_after](/reference/resource-configs/build-after): <dict> # Only for state aware orchestration on dbt Enterprise or Enteprise+ plans.
```

>     [+](/reference/resource-configs/plus-prefix)[build_after](/reference/resource-configs/build-after): <dict> # Only for state aware orchestration in the dbt plaftorm 

Reviewer: This link didn't work and I believe this should link to freshness.
```suggestion
    [+](/reference/resource-configs/plus-prefix)[freshness](/reference/resource-configs/freshness): <dict> 
```

>     [+](/reference/resource-configs/plus-prefix)[freshness](/reference/resource-configs/freshness): <dict> # Only for state aware orchestration on dbt platform Enterprise tiers.

Reviewer: I don't think Freshness is limited like this, only build_after 
```suggestion
    [+](/reference/resource-configs/plus-prefix)[freshness](/reference/resource-configs/freshness): <dict>
```

Author: Ah good catch!

>     [+](/reference/resource-configs/plus-prefix)[freshness](/reference/resource-configs/freshness): <dict> # Only for state aware orchestration on dbt Enterprise or Enteprise+ plans.

Author: ```suggestion
    [+](/reference/resource-configs/plus-prefix)[freshness](/reference/resource-configs/freshness): <dict> # Only for state aware orchestration on dbt platform Enterprise tiers.
```

>     [+](/reference/resource-configs/plus-prefix)[build_after](/reference/resource-configs/build-after): <dict> # Only for state aware orchestration on dbt Enterprise or Enteprise+ plans.

Author: ```suggestion
    [+](/reference/resource-configs/plus-prefix)[freshness](/reference/resource-configs/freshness): <dict> # Only for state aware orchestration on dbt Enterprise or Enteprise+ plans.
```

  - name: [<model-name>]
    config:
      freshness:

>         # build_after available for dbt platform only and is nested under freshness

Author: ```suggestion
        # build_after is available for dbt Enterprise and Enterprise+ plans only and is nested under freshness.
```

>         # build_after is nested under freshness. Available on dbt platform Enterprise tiers only.

Reviewer 2: this config is available in core as well, it's just only _used_ for SAO!

>         # build_after is available for dbt Enterprise and Enterprise+ plans only and is nested under freshness.

Author: ```suggestion
        # build_after is nested under freshness. Available on dbt platform Enterprise tiers only.
```

  - name: stg_orders
    config:
      freshness:

>         build_after:  # build this model no more often than every X amount of time, as long as as it has new data. Available only for state aware orchestration on the dbt platform. 

Author: ```suggestion
        build_after:  # build this model no more often than every X amount of time, as long as it has new data. Available only on dbt platform Enterprise tiers. 
```
